### PR TITLE
Add new tool for bugsnag list project collaborator

### PR DESCRIPTION
**Pull request by SB-gulshans**
## **Goal**

Add a new tool to retrieve project collaborators from BugSnag, enabling users to list all team members with access to a specific project.

## Design

The implementation follows the existing BugSnag tool architecture:

1. **API Layer**: Extended `ProjectAPI` with a new `listProjectCollaborators()` method that calls the `/projects/{project_id}/collaborators` endpoint and supports optional filtering by collaborator name or ID
2. **Tool Layer**: Created `ListProjectCollaborators` tool class that:
   - Accepts `projectId` (required) and optional `collaboratorNameOrId` filter parameter
   - Supports partial name matching and exact ID matching
   - Returns results with metadata (count and filtered flag)
3. **Type Definitions**: Added `CollaboratorApiView` and `Collaborator` interfaces to define the API response structure with fields like name, email, admin status, etc.

## Changeset

**Files modified:**
- `CHANGELOG.md` — Added entry documenting the new tool
- `src/bugsnag/client.ts` — Registered the new `ListProjectCollaborators` tool
- `src/bugsnag/client/api/Project.ts` — Implemented `listProjectCollaborators()` API method with filtering
- `src/bugsnag/client/api/api.ts` — Added `CollaboratorApiView` interface and `ProjectsApiFetchParamCreator.listProjectCollaborators()` fetch configuration
- `src/bugsnag/client/api/index.ts` — Exported `Collaborator` type
- `src/bugsnag/tool/project/list-project-collaborators.ts` — New tool implementation (79 lines)
- `src/bugsnag/utils/factories.ts` — Added `getMockCollaborator()` test factory
- `src/common/__snapshots__/server-definitions.test.ts.snap` — Updated tool registry snapshot
- `src/bugsnag/client.test.ts` — Added 10 comprehensive test cases covering success, filtering, empty results, error handling, and edge cases

**Test Coverage:**
- Lists all collaborators for a project
- Filters collaborators by name (partial match)
- Filters collaborators by ID (exact match)
- Handles empty results with appropriate messages
- Gracefully handles API errors
- Handles null/undefined responses
- Validates required parameters
- Handles large datasets (100+ collaborators)

## Testing

The implementation includes **10 test cases** in `client.test.ts` that validate:
- Normal operation with multiple collaborators
- Name-based filtering (case-insensitive partial match)
- ID-based filtering (exact match)
- Empty result sets (unfiltered and filtered)
- API error handling with descriptive error messages
- Null response handling
- Required parameter validation
- Large-scale datasets (100 collaborators)

All tests mock the `ProjectAPI.listProjectCollaborators()` method and verify correct API calls and response formatting.

### CODE REVIEW

File: src/bugsnag/tool/project/list-project-collaborators.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      return {
+        content: [{ type: "text", text: JSON.stringify(content) }],
+      };
+    } catch (error) {
+      console.warn("Failed to fetch project collaborators:", error);
+      throw new ToolError(
+        `Failed to list collaborators for project ${params.projectId}: ${
+          error instanceof Error ? error.message : String(error)
+        }`,
+      );
```
[the remark this answers is not in the document]
Comment (SB-gulshans) [reply src/bugsnag/tool/project/list-project-collaborators.ts:70]: Not required
